pkg/geo/geogfn/geohash.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geogfn
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/cockroachdb/cockroach/pkg/geo/geopb"
+	"github.com/cockroachdb/errors"
+)
+
+// GeoHashNoPrecision is used to calculate the precision based on
+// the size of the feature bounding box.
+const GeoHashNoPrecision = 0
+
+// GeoHash returns the string representing the geohash from the feature a based on
+// the precision provided p. If no precision is given, it is calculated from
+// the bounding box size: larger features have less precision; smaller features have more.
+// For point features, a default precision of 20 is used.
+func GeoHash(a *geo.Geography, p int) (string, error) {
+
+	bb := a.SpatialObject().BoundingBox
+	if bb == nil {
+		return "", nil
+	}
+
+	// GeoHash needs decimal degrees:
+	if bb.MinX < -180.0 || bb.MinY < -90.0 || bb.MaxX > 180.0 || bb.MaxY > 90.0 {
+		return "", errors.Newf("GeoHash requires inputs in decimal degrees, got (%f, %f, %f, %f)", bb.MinX, bb.MinY, bb.MaxX, bb.MaxY)
+	}
+
+	// It uses the size of the bounding box to calculate the precision
+	if p <= GeoHashNoPrecision {
+		p = getPrecisionByBBox(bb)
+	}
+
+	// returns the center from the bounding box to calculate the GeoHash
+	bbCenterLng := bb.MinX + (bb.MaxX-bb.MinX)/2.0
```

Comment:
if you rebase on top of the PR mentioned, the center of the bounding box is actually `a.BoundingRect().Centroid()` (we're operating on a sphere!)